tests/test_flows.py

```diff
@@ -814,53 +834,131 @@ def test_client_capability_query(self):
+                                            tlv(0x90, 0x000C, '{} {}'.format(
+                                                env.agents[0].radios[0].mac, sta2.mac)))
         time.sleep(1)
 
-        debug("Confirming client capability report message has been received on controller")
-        self.check_log(env.controller, r"Received CLIENT_CAPABILITY_REPORT_MESSAGE")
-        self.check_log(env.controller,
-                       r"Result Code= SUCCESS, client MAC= {}, BSSID= {}"
-                       .format(sta2.mac, env.agents[0].radios[0].mac))
-
-        for line in association_frame.splitlines():
-            self.check_log(env.controller, r"{}".format(line))
-
+        query2 = self.check_cmdu_type_single("client capability query", 0x8009,
+                                             env.controller.mac, env.agents[0].mac, mid2)
+        if query2:
+            query2_tlv = self.check_cmdu_has_tlv_single(query2, 0x90)
+            if query2_tlv:
+                try:
+                    if query2_tlv.client_info_mac_addr != sta2.mac:
+                        self.fail("Wrong mac address in query")
+                except AttributeError:
+                    self.fail("Query does not contain mac address")
+                try:
+                    if query2_tlv.client_info_bssid != env.agents[0].radios[0].mac:
+                        self.fail("Wrong bssid in query")
+                except AttributeError:
+                    self.fail("Query does not contain bssid")
+
+        report2 = self.check_cmdu_type_single("client capability report", 0x800a,
+                                              env.agents[0].mac, env.controller.mac, mid2)
+        if report2:
+            report2_client_info_tlv = self.check_cmdu_has_tlvs(report2, 0x90)[0]
+            if report2_client_info_tlv:
+                try:
+                    if report2_client_info_tlv.client_info_mac_addr != sta2.mac:
+                        self.fail("Wrong mac address in report")
+                except AttributeError:
+                    self.fail("Report does not contain mac address")
+                try:
+                    if report2_client_info_tlv.client_info_bssid != env.agents[0].radios[0].mac:
+                        self.fail("Wrong bssid in report")
+                except AttributeError:
+                    self.fail("Report does not contain bssid")
+
+            report2_cap_report_tlv = self.check_cmdu_has_tlvs(report2, 0x91)[0]
+            if report2_cap_report_tlv:
+                try:
+                    if int(report2_cap_report_tlv.client_capability_result, 0) != 0x00:
+                        self.fail("Capability report result is not successful")
+                    if report2_cap_report_tlv.client_capability_frame != association_frame:
+                        self.fail("Capability report does not contain correct frame")
+                        debug(f"Frame received\n{report2_cap_report_tlv.client_capability_frame}")
```

Comment:
Oh, if most of the code is written that way I can change it too. But did you consider using format strings instead of .format() ? Many people (for example devs in my  current company :)) like format strings more.

Reply:
Then I'll definitely remove all format strings.